File: LENGUAJES/PYTHON/Ejercicio13.py
#OCTAVOS DE FINAL

import random

def simular_partido(equipo1, equipo2):
    #Simula un partido entre dos equipos y devuelve el ganador.
    resultado = random.choice([equipo1, equipo2])
    return resultado

def octavos_de_final(equipos):
    #Simula los octavos de final con los equipos dados.
    print("Octavos de Final:")
    ganadores = []

    # Asegurarse de que hay un número par de equipos
    if len(equipos) % 2 != 0:
        print("El número de equipos debe ser par.")
        return

    for i in range(0, len(equipos), 2):
        equipo1 = equipos[i]
        equipo2 = equipos[i + 1]
        ganador = simular_partido(equipo1, equipo2)
        ganadores.append(ganador)
        print(f"Partido: {equipo1} vs {equipo2} - Ganador: {ganador}")

    return ganadores

def main():
    # Definimos los equipos que participan en los octavos de final
    equipos = [
        "FC Barcelona",
        "Real Madrid",
        "Arsenal",
        "Manchester City",
        "Chelsea",
        "Liverpool",
        "PSG",
        "Benfica"
    ]

    # Simular los octavos de final
    ganadores = octavos_de_final(equipos)

    print("\nEquipos que avanzan a los cuartos de final:")
    for ganador in ganadores:
        print(ganador)

if __name__ == "__main__":
    main()

File: LENGUAJES/JAVASCRIPT/Ejercicio13.js
// Octavos de final

function simularPartido(equipo1, equipo2) {
    // Simula un partido entre dos equipos y devuelve el ganador.
    return Math.random() < 0.5 ? equipo1 : equipo2;
}

function octavosDeFinal(equipos) {
    // Simula los octavos de final con los equipos dados.
    console.log("Octavos de Final:");
    const ganadores = [];

    // Asegurarse de que hay un número par de equipos
    if (equipos.length % 2 !== 0) {
        console.log("El número de equipos debe ser par.");
        return;
    }

    for (let i = 0; i < equipos.length; i += 2) {
        const equipo1 = equipos[i];
        const equipo2 = equipos[i + 1];
        const ganador = simularPartido(equipo1, equipo2);
        ganadores.push(ganador);
        console.log(`Partido: ${equipo1} vs ${equipo2} - Ganador: ${ganador}`);
    }

    return ganadores;
}

function main() {
    // Definimos los equipos que participan en los octavos de final
    const equipos = [
        "FC Barcelona",
        "Real Madrid",
        "Arsenal",
        "Manchester City",
        "Chelsea",
        "Liverpool",
        "PSG",
        "Benfica"
    ];

    // Simular los octavos de final
    const ganadores = octavosDeFinal(equipos);

    console.log("\nEquipos que avanzan a los cuartos de final:");
    for (const ganador of ganadores) {
        console.log(ganador);
    }
}
main();